src/plugin/dl/dlwrappers.cpp

```diff
@@ -71,29 +72,41 @@ getRpathRunPath(void *caller, string *rpathStr, string *runpathStr)
   ASSERT_NOT_NULL(strtab);
 
   if (rpath != NULL) {
-    *rpathStr = strtab + rpath->d_un.d_val;
-    *rpathStr = dmtcp::Util::replace(*rpathStr, "$ORIGIN", dirname);
+    Util::replace(rpathStr, &strtab[rpath->d_un.d_val], "$ORIGIN", dirname);
   } else {
-    rpathStr->clear();
+    rpathStr[0] = '\0';
   }
 
   if (runpath != NULL) {
-    *runpathStr = strtab + runpath->d_un.d_val;
-    *runpathStr = dmtcp::Util::replace(*runpathStr, "$ORIGIN", dirname);
+    Util::replace(runpathStr, &strtab[runpath->d_un.d_val], "$ORIGIN", dirname);
```

Comment:
'man ld.so' says that `${ORIGIN}` is also valid.  We should also test for that.

Reply:
Addressed `${ORIGIN}` and added asserts for $LIB, etc.